meson.build

```diff
@@ -0,0 +1,1554 @@
+    'use': false,
+}
+if x11['use']
+    if gl_x11['deps'].found() and get_option('gl')
+       gl_x11 += {'use': true}
+    endif
+endif
+if gl_x11['use']
+    dependencies += gl_x11['deps']
+    sources += 'video/out/opengl/context_glx.c'
+endif
+
+egl = {
+    'name': 'egl',
+    'deps': dependency('egl', version: '> 1.4.0', required: get_option('egl')),
+    'use' : false
+}
+if egl['deps'].found() and not get_option('gl').disabled()
+    dependencies += egl['deps']
+    egl += {'use': true}
+endif
+
+egl_helpers = {
+    'name': 'egl_helpers',
+    'use': egl['use'] # or rpi or egl_angle_win32 or egl_android
+}
+if egl_helpers['use']
+    sources += 'video/out/opengl/egl_helpers.c'
+endif
+
+if egl['use'] and egl_helpers['use']
+    sources += 'video/filter/vf_gpu.c'
+endif
+
+egl_drm = {
+    'name': 'egl-drm',
+    'use': drm['use'] and gbm['use'] and egl['use'] and not get_option('egl-drm').disabled(),
+}
+if egl_drm['use']
+    sources += 'video/out/opengl/context_drm_egl.c'
+endif
+
+gl_wayland = {
+    'name': 'gl-wayland',
+    'deps': dependency('wayland-egl', version: '>= 9.0.0', required: get_option('gl-wayland')),
+}
+gl_wayland += {'use': wayland['use'] and egl['use'] and gl_wayland['deps'].found()}
+if gl_wayland['use']
+    dependencies += gl_wayland['deps']
+    sources += 'video/out/opengl/context_wayland.c'
+endif
+
+gl_win32_fragment = run_command('cat', 'waftools/fragments/gl_win32.c')
+gl_win32 = {
+    'name': 'gl-win32',
+    'deps': cc.find_library('opengl32', required: get_option('gl-win32')),
+    'statement': gl_win32_fragment.stdout(),
+    'use': false,
+}
+if gl_win32['deps'].found() and win32
+    gl_win32 += {'use': cc.compiles(gl_win32['statement'], dependencies: gl_win32['deps'], name: 'gl-win32 check')}
+endif
+if gl_win32['use']
+    dependencies += gl_win32['deps']
+    sources += 'video/out/opengl/context_win.c'
+endif
+
+gl_dxinterop = {
+    'name': 'gl-dxinterop',
+    'd3d_statement': run_command('cat', 'waftools/fragments/gl_dxinterop_d3d.c').stdout(),
+    'gl_statement': run_command('cat', 'waftools/fragments/gl_dxinterop_gl.c').stdout(),
+    'use': false,
+}
+if gl_win32['use'] and not get_option('gl-dxinterop').disabled()
+    gl_dxinterop_d3d = cc.compiles(gl_dxinterop['d3d_statement'], name: 'gl-dxinterop d3d check')
+    gl_dxinterop_gl = cc.compiles(gl_dxinterop['gl_statement'], name: 'gl-dxinterop gl check')
+    gl_dxinterop += {'use': gl_dxinterop_d3d and gl_dxinterop_gl}
+endif
+if gl_dxinterop['use']
+    sources += 'video/out/opengl/context_dxinterop.c'
+endif
+
+egl_x11 = {
+    'name': 'egl-x11',
+    'use': x11['use'] and egl['use'] and not get_option('egl-x11').disabled(),
+}
+if egl_x11['use']
+    sources += 'video/out/opengl/context_x11egl.c'
+endif
+
+if egl_x11['use'] or gl_x11['use']
+    sources += 'video/out/opengl/oml_sync.c'
+endif
+
+gl = {
+    'name': 'gl',
+    'use': gl_x11['use'] or egl_x11['use'] or egl_drm['use'] or gl_wayland['use'] or get_option('plain-gl')
```

Comment:
Meson throw `ERROR: Second argument to "or" is not a boolean.` because default value for `get_option('plain-gl')` is auto

Reply:
I'm probably going to rewrite all this gl stuff.